### PR TITLE
Add eviction feature

### PR DESCRIPTION
**Pull request by alyiox**

### CODE REVIEW

File: src/DotQuery.Extensions/ThreadSafeQueryCache.cs [hunk 1]
```diff
@@ -45,5 +46,10 @@ public TResult GetOrAdd(TKey key, TResult lazyTask)
         }
 
         public TResult GetOrAdd(TKey key, TResult lazyTask, EntryOptions options) => GetOrAdd(key, lazyTask);
+
+        public void Remove(TKey key)
+        {
+            throw new NotSupportedException();
```
Comment (karldodd) [inline src/DotQuery.Extensions/ThreadSafeQueryCache.cs]: Why not supported here?

File: src/DotQuery.Core/Caches/EmptyQueryCache.cs [hunk 2]
```diff
@@ -45,5 +47,9 @@ public TValue GetOrAdd(TKey key, TValue lazyTask, EntryOptions options)
         public void Set(TKey key, TValue value, EntryOptions options)
         {
         }
+
+        public void Remove(TKey key)
```
Comment (karldodd) [inline src/DotQuery.Core/Caches/EmptyQueryCache.cs:51]: I guess we should return bool to indicate whether an item is really removed.

Comment (alyiox) [inline src/DotQuery.Core/Caches/EmptyQueryCache.cs:51]: Mee too. However the `Microsoft.Extensions.Caching.Memory.IMemoryCache` just has a method with the signature `void Remove(object key)` and doesn't return `boolean`.

File: src/DotQuery.Extensions/ThreadSafeQueryCache.cs (hunk 1 above)
Comment (alyiox) [inline src/DotQuery.Extensions/ThreadSafeQueryCache.cs]: Fixed.

**Pull request merged**